launch/cli/endpoints.py

```diff
@@ -119,3 +123,129 @@ def get_endpoint(ctx: click.Context, endpoint_name: str):
+            )
+
+        post_inference_hooks_choices = []
+        post_inference_hooks = model_endpoint.post_inference_hooks or []
+        for hook in PostInferenceHooks:
+            value = hook.value  # type: ignore
+            post_inference_hooks_choices.append(
+                q.Choice(title=value, checked=(value in post_inference_hooks))
+            )
+
+    if model_endpoint.status != "READY":
+        pretty_print(
+            f"Endpoint '{endpoint_name}' is not ready. Please wait for it to be ready "
+            "before editing."
+        )
+        return
+
+    model_bundle = q.select(
+        "Model bundle: ", choices=model_bundle_choices
+    ).ask()
+    resource_state = _dict_not_none_or_empty(
+        **(model_endpoint.resource_state or {})
+    )
+    deployment_state = _dict_not_none_or_empty(
+        **(model_endpoint.deployment_state or {})
+    )
+    cpus = q.text("Cpus: ", default=resource_state.get("cpus", "")).ask()
+    gpu_raw = q.text(
+        "Gpus: ",
+        default=str(resource_state.get("gpus", "")),
+        validate=_validate_int,
+    ).ask()
+    gpus = int(gpu_raw)
+    memory = q.text("Memory: ", default=resource_state.get("memory", "")).ask()
+    storage = q.text(
+        "Storage (optional): ", default=resource_state.get("storage", "")
+    ).ask()
+    gpu_type_prompt = "Gpu type (optional): " if gpus == 0 else "Gpu type: "
+    gpu_type = q.select(
+        gpu_type_prompt,
+        choices=["", "nvidia-tesla-t4", "nvidia-ampere-a10", "nvidia-a100"],
+    ).ask()
+    min_workers = q.text(
+        "Min workers: ",
+        default=str(deployment_state.get("min_workers", "")),
+        validate=_validate_int,
+    ).ask()
+    min_workers = int(min_workers)
+    max_workers = q.text(
+        "Max workers: ",
+        default=str(deployment_state.get("max_workers", "")),
+        validate=_validate_int,
+    ).ask()
+    max_workers = int(max_workers)
+    per_worker = q.text(
+        "Per worker: ",
+        default=str(deployment_state.get("per_worker", "")),
+        validate=_validate_int,
+    ).ask()
+    per_worker = int(per_worker)
+    post_inference_hooks = q.checkbox(
+        "Post-inference hooks: ", choices=post_inference_hooks_choices
+    ).ask()
+    default_callback_url = q.text(
+        "Default callback url (optional): ",
+        default=model_endpoint.default_callback_url or "",
+    ).ask()
+
+    kwargs = _dict_not_none_or_empty(
+        model_bundle=model_bundle,
+        cpus=cpus,
+        memory=memory,
+        storage=storage,
+        gpus=gpus,
+        min_workers=min_workers,
+        max_workers=max_workers,
+        per_worker=per_worker,
+        gpu_type=gpu_type,
+        post_inference_hooks=post_inference_hooks,
+        default_callback_url=default_callback_url,
+    )
+
+    with spinner(f"Editing endpoint '{endpoint_name}'"):
+        client.edit_model_endpoint(model_endpoint=model_endpoint, **kwargs)
```

Comment:
do we want(/need) to print out a nice error message if the user passes in parameters that end up failing server-side validation? (or does the error message seem good enough)

Reply:
Yeah that would be a good TODO - error messages are hard 😢